### PR TITLE
Contract test for tracing propagation in serverless 

### PR DESCRIPTION
**Pull request by MichalKalke**
**Description**

Function runtimes must propagate incoming tracing headers so that when tracing is enabled the functions do not break the traceable request chain.

Changes proposed in this pull request:

- add tracing scenario
- add test for nodejs18
- add test for nodejs18
- add make target

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/17507

### CODE REVIEW

File: tests/function-controller/testsuite/scenarios/tracing.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+	genericContainer := shared.Container{
+		DynamicCli:  dynamicCli,
+		Namespace:   cfg.Namespace,
+		WaitTimeout: cfg.WaitTimeout,
+		Verbose:     cfg.Verbose,
+		Log:         logf,
+	}
+
+	//python39Fn := function.NewFunction("python39", cfg.KubectlProxyEnabled, genericContainer.WithLogger(python39Logger))
```
Comment (dbadura) [inline tests/function-controller/testsuite/scenarios/tracing.go:50]: Python 39 will be added in another PR

**Review by pPrecel — APPROVED** [review]
